depdive/src/update.rs

```diff
@@ -170,18 +176,63 @@ impl UpdateAnalyzer {
             Self::get_version_change_info_from_summarydiff(&summary_id, &summary_diff_status);
         let repository = Self::get_repository_from_graphs(&vec![prior_graph, post_graph], &name);
 
+        // A temporary holders for build_paths to check build script at both versions
+        let mut build_script_paths: Vec<String> = Vec::new();
         let old_version = version_change_info.old_version;
+        if !old_version.is_none() {
+            build_script_paths
+                .extend(Self::get_build_script_paths(prior_graph, &name)?.unwrap_or(vec![]));
+        }
         let new_version = version_change_info.new_version;
+        if !new_version.is_none() {
+            build_script_paths
+                .extend(Self::get_build_script_paths(post_graph, &name)?.unwrap_or(vec![]));
+        }
+
+        let build_script_paths = if build_script_paths.is_empty() {
```

Comment:
Any particular reason you want `Some(HashSet)` instead of just allowing an empty HashSet?

You could allow empty hashset instead of none, and then just build it this way:

```rust
let build_script_paths: HashSet<_> = build_script_paths.iter().cloned().collect();
```

Reply:
I just thought it might be more meaningful to the user getting `None` than an empty set. I do agree it makes for bloated code and I eventually had to write more for this.